server/sonar-server/src/main/java/org/sonar/server/organization/OrganizationCreationImpl.java

```diff
@@ -69,6 +77,42 @@ public OrganizationDto create(DbSession dbSession, long creatorUserId, NewOrgani
     return organization;
   }
 
+  @Override
+  public Optional<OrganizationDto> createForUser(DbSession dbSession, UserDto newUser) {
+    if (!isCreatePersonalOrgEnabled()) {
+      return Optional.empty();
+    }
+
+    NewOrganization newOrganization = newOrganizationBuilder()
+      .setKey(organizationValidation.generateKeyFrom(newUser.getLogin()))
+      .setName(toName(newUser.getLogin()))
+      .build();
+    checkState(!organizationKeyIsUsed(dbSession, newOrganization.getKey()),
+      "Can't create organization with key '%s' for new user '%s' because an organization with this key already exists",
```

Comment:
nop, this is an unsupported state as creation of personal orgs should actually never fail.
We should really implement a smarter algorithm to generate the key but no time just now. We'd rather just consider this case as not supported (yet)